src/base64-inl.h

```diff
@@ -136,29 +153,50 @@ inline size_t base64_encode(const char* src,
   unsigned c;
   unsigned i;
   unsigned k;
-  unsigned n;
+  unsigned n32;
+  unsigned n64;
 
   const char* table = base64_select_table(mode);
 
   i = 0;
   k = 0;
-  n = slen / 3 * 3;
+  n32 = (slen >= 4 ? ((slen - 1) / 3) * 3 : 0);
+  n64 = (slen >= 8 ? ((slen - 2) / 6) * 3 : 0);
+
+  // Read in chunks of 8 bytes for as long as possible
+  while (i < n64) {
+    const uint64_t dword = *reinterpret_cast<const uint64_t*>(src + i);
```

Comment:
Some architectures don't support unaligned 64-bit access, so this could end up being somewhat slower on those. But still an optimization on most systems!

Reply:
That's why I was curious to check on ARM. However from what I'm seeing ARMv7 and newer support unaligned reads, with ARMv7 supposedly not supporting them for only 2 instructions?

Reply:
Have you tried to start reading 1/4 bytes until 8 byte alignment is reached?